packages/calcite-components/src/components/button/button.scss

```diff
@@ -685,6 +685,16 @@
   @apply text-0h w-11;
   min-block-size: theme("height.11");
 }
+
+:host([scale="s"][width="full"]) button:not(.content--slotted),
+:host([scale="s"][width="full"]) a:not(.content--slotted),
+:host([scale="m"][width="full"]) button:not(.content--slotted),
+:host([scale="m"][width="full"]) a:not(.content--slotted),
+:host([scale="l"][width="full"]) button:not(.content--slotted),
+:host([scale="l"][width="full"]) a:not(.content--slotted) {
+  @apply w-full;
+}
```

Comment:
alternative approach

```css
:host(:is([scale="s"], [scale="m"], [scale="l"])[width="full"]) {
  a, button {
    &:not(.content--slotted) {
      inline-size: var(--calcite-container-size-content-fluid);
    }
  }
}
```

Reply:
Another alternative approach would be to apply a class here using JS since the CSS logic is getting a little complex here.